sdk/include/platform/sunburst/v0.2/platform-gpio.hh

```diff
@@ -0,0 +1,143 @@
+#pragma once
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * Represents the state of the Sonata's joystick.
+ *
+ * Note, that up to three of the bits may be asserted at any given time.
+ * There may be up to two cardinal directions asserted when the joystick is
+ * pushed in a diagonal and the joystick may be pressed while being pushed in a
+ * given direction.
+ */
+enum class SonataJoystick : uint8_t
+{
+	Left    = 1 << 0,
+	Up      = 1 << 1,
+	Pressed = 1 << 2,
+	Down    = 1 << 3,
+	Right   = 1 << 4,
+};
+
+/**
+ * A Simple Driver for the Sonata's GPIO.
+ *
+ * Documentation source can be found at:
+ * https://github.com/lowRISC/sonata-system/blob/1a59633d2515d4fe186a07d53e49ff95c18d9bbf/doc/ip/gpio.md
+ *
+ * Rendered documentation is served from:
+ * https://lowrisc.org/sonata-system/doc/ip/gpio.html
+ */
+struct SonataGPIO
+{
+	uint32_t output;
+	uint32_t input;
+	uint32_t debouncedInput;
+	uint32_t debouncedThreshold;
+	uint32_t raspberryPiHeader;
+	uint32_t raspberryPiMask;
+	uint32_t arduinoShieldHeader;
+	uint32_t arduinoShieldMask;
+
+	/**
+	 * Is the current target Sonata 1.0.0 or later?
+	 */
+	static bool constexpr Sonata1OrLater =
+#if SONATA >= CHERIOT_VERSION_TRIPLE(1, 0, 0)
+	  true
+#else
+	  false
+#endif
```

Comment:
This probably isn't needed since this is just 0.2 now.

Reply:
Done. I haven't removed `CHERIOT_VERSION_TRIPLE`. It isn't used anywhere else that I could see. Should I remove it?

Reply:
I think it's fine to keep `CHERIOT_VERSION_TRIPLE` as we might need it for other things in future and we still define `SONATA` in that format.